compose/cli/log_printer.py

```diff
@@ -41,7 +42,7 @@ def no_color(text):
         for color_func, container in zip(color_funcs, self.containers):
             generator_func = get_log_generator(container)
             prefix = color_func(build_log_prefix(container, prefix_width))
-            yield generator_func(container, prefix, color_func)
+            yield generator_func(container, prefix, color_func, self.cascade_stop)
```

Comment:
OK, then I don't push this right now, but will do tests and implement this 'case 3' also.